components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -439,12 +443,28 @@ private OAuthConsumerAppDTO getApplicationById(String clientId, boolean isApplic
     }
 
     private Application createOAuthApplication(ApplicationRegistrationRequest registrationRequest)
-            throws DCRMException {
+            throws DCRMException, UserStoreException {
 
         String applicationOwner = StringUtils.isNotBlank(registrationRequest.getExtApplicationOwner()) ?
                 registrationRequest.getExtApplicationOwner() :
                 PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+
         String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        /*
+        ApplicationOwner will be null and a server error is thrown when creating an app, if the api authentication/
+        api security is disabled for DCR endpoint.
+        In such cases, we set the tenant admin as the owner of the application.
+        */
+        if (StringUtils.isBlank(applicationOwner)) {
+            DCRConfiguration dcrConfiguration = DCRConfigUtils.getDCRConfiguration();
+            boolean isClientAuthenticationRequired = dcrConfiguration.isAuthenticationRequired() != null ?
+                    dcrConfiguration.isAuthenticationRequired() : true;
+            if (!isClientAuthenticationRequired) {
+                applicationOwner = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUserRealm()
```

Comment:
Is there any limitation on setting the application owner's name in the DCRConfigurations even if the client authentication is not required. This fallback will lead to incorrect data.

Reply:
Can you explain what is meant by `"setting the application owner's name in the DCRConfigurations"` ? Do you meant having a constant value for the `applicationOwner` if authentication is disabled?

If further explaining, if `applicationOwner` is null, a server error is thrown now. So far this has not been an issue since the DCR was an authenticated endpoint and we can resolve the user and set to `applicationOwner` with the auth data. Now with the ability to disable authentication for DCR, since we cannot resolve a user, we take the tenant admin as the `applicationOwner`.

Reply:
@DMHP hope this is clear